node.c

```diff
@@ -0,0 +1,74 @@
+#include "node.h"
+#include "board.h"
+
+
+void initializeRoot(node* root, struct board *boardState){
+
+    // Setting default values
+    root->parent = NULL;
+    root->childMoves = (int *)malloc(2 * sizeof(int));
```

Comment:
Let's give this an initial size of 4.

 - [ ] Give childMoves an initial size of 4 in initialize()